gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java

```diff
@@ -66,16 +292,55 @@ private String parseFromAsid(COPCIN000001UK01Message payload) {
             .get(0)
             .getExtension();
     }
-
     private String parseToAsid(COPCIN000001UK01Message payload) {
         return payload.getCommunicationFunctionSnd()
             .getDevice()
             .getId()
             .get(0)
             .getExtension();
     }
-
     private String parseMessageRef(COPCIN000001UK01Message payload) {
         return payload.getId().getRoot();
     }
+    public boolean sendNackMessage(NACKReason reason, COPCIN000001UK01Message payload, String conversationId) {
```

Comment:
maybe NACK/ACK preparation stuff can be pulled into their own service/handler (not sure myself) since the COPC Handler is very long ?
if this logic is specific to COPC, which i don't *think* it is, it would need to be a CopcNackMessageHandler I think.